apps/root/src/pages/earn/components/strategies-table/index.tsx

```diff
@@ -0,0 +1,267 @@
+            {column.getOrderValue ? (
+              <TableSortLabel
+                active={orderBy === column.key}
+                direction={orderBy === column.key ? order : 'asc'}
+                onClick={() => onRequestSort(column.key)}
+              >
+                <StyledBodySmallLabelTypography>{column.label}</StyledBodySmallLabelTypography>
+              </TableSortLabel>
+            ) : (
+              <StyledBodySmallLabelTypography>{column.label}</StyledBodySmallLabelTypography>
+            )}
+          </TableCell>
+        ))}
+        <StyledTableEnd size="small"></StyledTableEnd>
+      </TableRow>
+    </TableHead>
+  );
+};
+
+const ROWS_PER_PAGE = 7;
+
+const skeletonRows = Array.from(Array(ROWS_PER_PAGE).keys());
+
+const AllStrategiesTableBodySkeleton = () => (
+  <>
+    {skeletonRows.map((i) => (
+      <TableRow key={i}>
+        {strategyColumnConfigs.map((col) => (
+          <TableCell key={col.key}>
+            {col.customSkeleton || (
+              <StyledBodySmallRegularTypo2>
+                <Skeleton variant="text" animation="wave" />
+              </StyledBodySmallRegularTypo2>
+            )}
+          </TableCell>
+        ))}
+        <StyledTableEnd size="small"></StyledTableEnd>
+      </TableRow>
+    ))}
+  </>
+);
+interface RowProps {
+  strategy: Strategy;
+  onRowClick: (strategy: Strategy) => void;
+}
+
+const renderBodyCell = (cell: React.ReactNode | string) =>
+  typeof cell === 'string' ? <StyledBodySmallRegularTypo2>{cell}</StyledBodySmallRegularTypo2> : cell;
+
+const Row = ({ strategy, onRowClick }: RowProps) => {
+  const [hovered, setHovered] = React.useState(false);
+
+  return (
+    <TableRow
+      key={strategy.id}
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+      sx={{ cursor: 'pointer' }}
+      hover
+      onClick={() => onRowClick(strategy)}
+    >
+      {strategyColumnConfigs.map((column) => (
+        <TableCell key={`${strategy.id}-${column.key}`}>{renderBodyCell(column.renderCell(strategy))}</TableCell>
+      ))}
+      <StyledTableEnd size="small">
+        <StyledNavContainer alignItems="center">
+          <DividerBorder2 orientation="vertical" />
+          <AnimatedChevronRightIcon $hovered={hovered} color="primary" />
+        </StyledNavContainer>
+      </StyledTableEnd>
+    </TableRow>
+  );
+};
+
+const createEmptyRows = (rowCount: number) => {
+  return Array.from({ length: rowCount }, (_, i) => (
+    <TableRow key={i} sx={{ visibility: 'hidden', height: ({ spacing }) => spacing(15.25) }}>
+      <TableCell colSpan={7}>&nbsp;</TableCell>
+    </TableRow>
+  ));
+};
+
+const AllStrategiesTable = ({
+  strategies,
+  hasFetchedAllStrategies,
+  defaultOrderBy,
+}: {
+  strategies: TableStrategy[];
+  hasFetchedAllStrategies: boolean;
+  defaultOrderBy: StrategyColumnKeys;
+}) => {
+  const [order, setOrder] = React.useState<Order>('desc');
```

Comment:
If we keep this on redux, the sorting will persist through the user changing pages/tabs, which is pretty cool if you ask me